actions/changelog-generate/action.yaml

```diff
@@ -0,0 +1,148 @@
+  merged-changelog:
+    description: "Changelog containing listing of all single changelogs"
+    value: ${{ steps.output-generator.outputs.merged-changelog }}
+runs:
+  using: "composite"
+  steps:
+    - name: "Set tags"
+      id: setTags
+      run: |
+        { # try
+          tag_list=$(git describe --tags)
+        } || { # catch
+          tag_list=""
+        }
+        # to-tag
+        if [[ "$tag_list" == *"${{ inputs.new-tag }}"* ]]; then
+          echo  "totag=${{ inputs.new-tag }}">> $GITHUB_OUTPUT
+          echo  "totag_exists=true">> $GITHUB_OUTPUT
+        else
+          echo  "totag=${{ github.event.repository.default_branch }}">> $GITHUB_OUTPUT 
+          echo  "totag_exists=false">> $GITHUB_OUTPUT
+        fi
+
+        #from-tag
+        setfrom=""
+        check_if_first_commit="false"
+
+        if [[ -n "${{ inputs.old-tag }}" ]] && [[ "$tag_list" == *"${{ inputs.old-tag }}"* ]]; then
+        # tag provided & provided tag exists
+            setfrom=${{ inputs.old-tag }}
+        else
+        # no tag provided or a non-existing tag was provided 
+          if [ -n "$tag_list" ]; then 
+            # if any tag exists then set from-tag to the latest tag
+            latest_tag=$(git describe --tags `git rev-list --tags --max-count=1`)
+            setfrom=$latest_tag
+          else
+            # if no tag exists then use the frist commit as fromtag
+            first_commit=`git rev-list --max-parents=0 HEAD`
+            setfrom=$first_commit
+            latest_commit=`git rev-parse HEAD`
+            echo  "totag=$latest_commit">> $GITHUB_OUTPUT 
+          fi
+        fi
+        echo  "fromtag=$setfrom">> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: "Create changelog"
+      id: build_changelog
+      uses: mikepenz/release-changelog-builder-action@v3.7.0
+      with:
+        token: "${{ inputs.github-token }}"
+        configuration: "${{ inputs.config }}"
+        fromTag: "${{steps.setTags.outputs.fromtag}}"
+        toTag: "${{steps.setTags.outputs.totag}}"
+        outputFile: "${{ inputs.output }}"
+        fetchReviewers: "${{ inputs.fetchReviewers }}"
+        fetchReleaseInformation: "${{ inputs.fetchReleaseInformation }}"
+        commitMode: "${{ inputs.commitMode }}"
+
+    - name: Replace default branch with new Tag
+      if: ${{ steps.setTags.outputs.totag_exists == 'false' }}
+      run: |
+        # Replace default branch with new tag
+        sed -i 's/\[${{steps.setTags.outputs.totag}}\]/\[${{ inputs.new-tag }}\]/g' ${{ inputs.output }}
+        sed -i 's|releases/tag/${{steps.setTags.outputs.totag}}|releases/tag/${{ inputs.new-tag }}|g' ${{ inputs.output }}
+
+        # Set Release date
+        tag_date=$(date '+%Y-%m-%d') 
+        sed -i "s/Release Date: /Release Date: \[${tag_date}\]/g" ${{ inputs.output }}
+      shell: bash
+
+    - name: Update github Changelog
+      id: output-generator
+      env:
+        tmp_changelog: tmp_changelog.md
```

Comment:
because we dont the action to do anything else then return the changelog. wether the changelog has to be updated should be decided outside of this action logic. that also explains why am using the `tmp_file`